apps/windows_explorer/windows_explorer.py

```diff
@@ -29,59 +91,380 @@
+}
+
+LANGUAGE_MAPPINGS = {
+    "en": {
+        "suffix_patterns": [
+            r" and [0-9]+ more tabs - File Explorer$",
+            r" and 1 more tab - File Explorer$",
+            r" - File Explorer$",
+        ],
+        "excludes": ["", "This PC", "File Explorer", "Gallery", "Home", "Network"],
+    },
+    "de": {
+        "suffix_patterns": [
+            r" und [0-9]+ weitere Registerkarten – Explorer$",
+            r" und 1 weitere Registerkarte – Explorer$",
+            r" – Datei-Explorer$",
+        ],
+        "excludes": ["", "Dieser PC", "Datei-Explorer", "Katalog", "Start", "Netzwerk"],
+    },
+}
+
 directories_to_remap = {}
-directories_to_exclude = {}
 
-if app.platform == "windows":
-    is_windows = True
-    import ctypes
 
-    GetUserNameEx = ctypes.windll.secur32.GetUserNameExW
-    NameDisplay = 3
-
-    size = ctypes.pointer(ctypes.c_ulong(0))
-    GetUserNameEx(NameDisplay, None, size)
-
-    nameBuffer = ctypes.create_unicode_buffer(size.contents.value)
-    GetUserNameEx(NameDisplay, nameBuffer, size)
-    one_drive_path = os.path.expanduser(os.path.join("~", "OneDrive"))
-
-    # this is probably not the correct way to check for onedrive, quick and dirty
-    if os.path.isdir(os.path.expanduser(os.path.join("~", r"OneDrive\Desktop"))):
-        directories_to_remap = {
-            "Desktop": os.path.join(one_drive_path, "Desktop"),
-            "Documents": os.path.join(one_drive_path, "Documents"),
-            "Downloads": os.path.join(user_path, "Downloads"),
-            "Music": os.path.join(user_path, "Music"),
-            "OneDrive": one_drive_path,
-            "Pictures": os.path.join(one_drive_path, "Pictures"),
-            "Videos": os.path.join(user_path, "Videos"),
-        }
-    else:
-        # todo use expanduser for cross platform support
-        directories_to_remap = {
-            "Desktop": os.path.join(user_path, "Desktop"),
-            "Documents": os.path.join(user_path, "Documents"),
-            "Downloads": os.path.join(user_path, "Downloads"),
-            "Music": os.path.join(user_path, "Music"),
-            "OneDrive": one_drive_path,
-            "Pictures": os.path.join(user_path, "Pictures"),
-            "Videos": os.path.join(user_path, "Videos"),
+# Utility functions
+def _string_to_guid(guid_string):
+    """Convert GUID string to ctypes GUID structure.
+
+    Args:
+        guid_string (str): GUID string in format '{XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX}'
+
+    Returns:
+        GUID: ctypes GUID structure
+    """
+
+    class GUID(ctypes.Structure):
+        _fields_ = [
+            ("Data1", ctypes.c_ulong),
+            ("Data2", ctypes.c_ushort),
+            ("Data3", ctypes.c_ushort),
+            ("Data4", ctypes.c_ubyte * 8),
+        ]
+
+    guid_string = guid_string.strip("{}").replace("-", "")
+    guid = GUID()
+    guid.Data1 = int(guid_string[0:8], 16)
+    guid.Data2 = int(guid_string[8:12], 16)
+    guid.Data3 = int(guid_string[12:16], 16)
```

Comment:
The Python `uuid.UUID` object can give you these with `.time_low`, `.time_mid` and `.time_hi_version`. I don't see a way to get 8 individual bytes so you'll have to do that yourself!